src/Kernel32.Desktop/Kernel32.cs

```diff
@@ -410,6 +409,172 @@ public enum SecurityImpersonationLevel
+        /// function to retrieve the full name of an executable image for both 32- and 64-bit processes from a 32-bit process.
+        /// </para>
+        /// </remarks>
+        [DllImport(nameof(Kernel32), SetLastError = true)]
+        public static extern SafeObjectHandle CreateToolhelp32Snapshot(
+            CreateToolhelp32SnapshotFlags dwFlags,
+            uint th32ProcessID);
+
+        /// <summary>Retrieves information about the first process encountered in a system snapshot.</summary>
+        /// <param name="hSnapshot">
+        /// A handle to the snapshot returned from a previous call to the
+        /// <see cref="CreateToolhelp32Snapshot" /> function.
+        /// </param>
+        /// <param name="lppe">
+        /// Contains process information such as the name of the executable file, the process identifier, and
+        /// the process identifier of the parent process.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true" /> if the first entry of the process list has been copied to the buffer or
+        /// <see langword="false" /> otherwise. The <see cref="Win32ErrorCode.ERROR_NO_MORE_FILES" /> error value is returned by
+        /// the <see cref="Marshal.GetLastWin32Error" /> function if no processes exist or the snapshot does not contain process
+        /// information.
+        /// </returns>
+        [DllImport(nameof(Kernel32), SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool Process32First(SafeObjectHandle hSnapshot, [In, Out] PROCESSENTRY32 lppe);
+
+        /// <summary>Retrieves information about the next process recorded in a system snapshot.</summary>
+        /// <param name="hSnapshot">
+        /// A handle to the snapshot returned from a previous call to the
+        /// <see cref="CreateToolhelp32Snapshot" /> function.
+        /// </param>
+        /// <param name="lppe">A <see cref="PROCESSENTRY32" /> structure.</param>
+        /// <returns>
+        /// Returns <see langword="true" /> if the next entry of the process list has been copied to the buffer or
+        /// <see langword="false" /> otherwise. The <see cref="Win32ErrorCode.ERROR_NO_MORE_FILES" /> error value is returned by
+        /// the <see cref="Marshal.GetLastWin32Error" /> function if no processes exist or the snapshot does not contain process
+        /// information.
+        /// </returns>
+        /// <remarks>
+        /// To retrieve information about the first process recorded in a snapshot, use the
+        /// <see cref="Process32First(SafeObjectHandle,PROCESSENTRY32)" />
+        /// function.
+        /// </remarks>
+        [DllImport(nameof(Kernel32), SetLastError = true, CharSet = CharSet.Unicode)]
+        public static extern bool Process32Next(
+            SafeObjectHandle hSnapshot,
+            [In, Out] PROCESSENTRY32 lppe);
```

Comment:
Very interesting. Yes, please add a section to the CONTRIB.md file describing this pattern and the things you've learned about it. 
As for struct vs. class in general, I think we should stick with struct for most cases since it means fewer allocations and depending on whose calling the method that may be very important. Using classes seems to make sense when the struct parameter is optional though since it makes it so much more natural and AFAIK there's no way we can avoid allocating some object or other in that case anyway.